litestar/contrib/sqlalchemy/base.py

```diff
@@ -99,6 +99,7 @@ def _sentinel(cls) -> Mapped[int]:
 class BigIntPrimaryKey:
     """BigInt Primary Key Field Mixin."""
 
+    id: Mapped[int]  # type: ignore[assignment] # pyright: ignore
```

Comment:
```suggestion
```
I think we'll want to remove this since it's handled by the `@declared_attr` decorator.  This is identical to how we handle the addition of the `_sentinel` column on the `UUID` mix-in

Reply:
When I get rid of that, I instead get an error:
`No type information found for 'BigIntBaseModel.id'. Has a type annotation been added to the column?` (on main).

Reply:
@MatthewNewland can you confirm that the sequences still get generated as you'd expect?  presumably they do?

Reply:
See: https://github.com/sqlalchemy/sqlalchemy/discussions/10008; I wasn't able to find a good answer through Googling.

I will try to setup Postgres to test sequences later today; I have pretty much just been using SQLite in the toy projects I've been building. I can confirm, though, that for SQLite it's working as expected!

Reply:
So re: the type:

- See https://github.com/sqlalchemy/sqlalchemy/discussions/10008
- It appears that the return type of a declared_attr is not going to show up in the __annotations__ dict of get_type_hints (or in native Python properties, descriptors generally?)
- There are a number of ways we could go about this. I am going to submit a commit that will just handle subclasses of BigIntPrimaryKey as a special case; then we won't be duplicating and will keep the linter happy. `id` is also such a core field it *seems* like special handling for it is OK, but we might want to think of something that handles declared_attrs better in DTOs generally. (I don't think that would be particularly easy, fwiw.)

Reply:
This is extremely helpful info.  Thanks for taking the time on this.